src/main/kotlin/org/malachite/estella/offer/api/OfferExceptionHandler.kt

```diff
@@ -13,10 +14,14 @@ import org.springframework.web.bind.annotation.ExceptionHandler
 class OfferExceptionHandler {
 
     @ExceptionHandler(OfferNotFoundException::class)
-    fun handleNoSuchElementException(ex: OfferNotFoundException): ResponseEntity<Any> =
-        ResponseEntity(Message("We couldn't find this offer"), HttpStatus.NOT_FOUND)
+    fun handleNoSuchElementException(ex: OfferNotFoundException): ResponseEntity<Message> =
+        OwnResponses.NO_RESOURCE("We couldn't find this offer")
 
     @ExceptionHandler(UnauthenticatedException::class)
-    fun handleUnauthenticated(ex: UnauthenticatedException): ResponseEntity<Any> =
+    fun handleUnauthenticated(ex: UnauthenticatedException): ResponseEntity<Message> =
         OwnResponses.UNAUTH
+
+    @ExceptionHandler(ApplicationAlreadyMadeOnOfferException::class)
+    fun handleUnauthenticated(ex: ApplicationAlreadyMadeOnOfferException): ResponseEntity<Any> =
+        OwnResponses.BAD_REQUEST("Someone already applied on this offer!")
```

Comment:
To nie jest dobry message XDD 
Ogólnie powinno być "someone with this e-mail already applied on this offer". Plus jest to trochę bug w authorization - jakby nie było, nie jesteśmy pewni, czy to faktycznie ta osoba zarekrutowala. Wydaje mi się, że powinniśmy sprawdzać, czy nie ma typa o takim koncie, jak aplikujemy przez no_user, ale tu też jest taki bug, że ten user i tak będzie. Ogólnie tak źle i tak niedobrze, ale ten message bym jednak zmienił.

(Możesz zignorować większość tego komentarza, jako ze to była moja wewnętrzna dyskusja ze sobą)

Reply:
Ale tu chodzi o case że nie można usunąć oferty gdy ktoś na nią zaaplikował XDD

Reply:
A co sobie bugi powaliłem XDDDD 
To sorry XDDDDD

Reply:
Ale i tak bym message zmienił na "You cannot delete this offer, because someone has...." bo to już będzie takie nawet bardziej pod front lepsze.